functions/functions.inc

```diff
@@ -442,3 +447,45 @@ function core_pages()
     return $valid;
 
 }
+
+
+function RBACApiRequest($rbac)
```

Comment:
Can you help me understand the motive behind moving this to the PHP server calling the API it is exposing itself? 

It seems like this is doing authorization with extra steps. Could we not just check that the session user has permission to a given area of the site and deny them based on that? Or is this specifically to handle the case that a user has an OAuth token that may have expired?

Or are we planning to move out authentication entirely so that it can function as its own service?

Reply:
The idea is to fully move all the RBAC code to the API side of the world.  From past experience, on other projects,  I always try to have layer separation enforced as much as I can, Just to help with possible API server being seperate from a possible load balanced front end, and yada yada. So I try to avoid calling directly into the API layer from the front end and instead do internal API calls.